### ISSUE TITLE
Issue with RE_TAGGED regex in Parser

### ISSUE THREAD

**Issue by surajwy** [mscdex/node-imap#537]
Hi Brian ( @mscdex ),

We found out that COPY, UID COPY command is hanging and its never coming back. Noticed that it was due to not matching 'OK' tagged response in the regex. 

Some server handles COPY, UID COPY command differently and only return a tagged response similar to this:  "**A6 OK**". Now, when we use RE_TAGGED regex to parse this response, we will never get any match as it does not contain any **textCode** or **text** part in it. Due to this mismatch, the COPY callback will never be called.

As per [RFC 3501](https://tools.ietf.org/html/rfc3501#section-6.4.7), only **(OK|NO|BAD)** are the required response words and rest depends on server implementation and UIDPlus extension.

I can think of two possible ways with which we can solve this issue.

**First way**: In Parser.js, function **_resTagged**, line 181, we can add another **else if** block like this -
```
// RE_TAGGED_OK = /^A(\d+) OK$/i
else if (m = RE_TAGGED_OK.exec(this._buffer)) {
    this._buffer = '';
    this.emit('tagged', {
      type: 'ok',
      tagnum: parseInt(m[1], 10)
    });
  }
   else
    this._buffer = '';
```
   This approach is kinda safe.

**Second way**: We can modify the RE_TAGGED regex itself to -
```
RE_TAGGED = /^A(\d+) (OK|NO|BAD)(?: )*(?:\[([^\]]+)\] )?(.*)$/i,
```

I have tested the first solution and it's working for those servers. 
Can you suggest, which solution is safe and should go to production? I'll code and send you a PR.

Thanks,
Suraj

**Comment by mscdex** [comment]
I think I would rather change the regexp. I believe the "zero or more spaces" could be tweaked to just support a single space and make it optional, like: `/^A(\d+) (OK|NO|BAD) ?(?:\[([^\]]+)\] )?(.*)$/i`. Code may/may not still need to be changed though.

If you do send in a PR, please make sure to add relevant parser tests.

**Comment by surajwy** [comment]
Thanks Brian for the closure. Have sent you a PR.

Regards,
Suraj

**Comment by mscdex** [comment]
Fixed by c2a859727de65dd5ff48b64e9c5c7c2f187b06cf.
